src/modules/auth/component/Login/Login.tsx

```diff
@@ -2,30 +2,62 @@ import {
   Grid,
 } from "@mui/material";
 import { ChangeEvent, FormEvent, useState } from "react";
+import { useAppDispatch, useAppSelector } from "../../../../store/hooks";
 import { PrimaryButton } from "../../../../components/UI/PrimaryButton";
 import { TextField } from "../../../../components/UI/TextField";
 import { Heading } from "../../../../components/UI/Heading";
-import { profilesData } from "../../../../data/data";
+import { login } from "../../../../store/features/tokenSlice";
+import { loginUser } from "../../../../store/features/userSlice";
+import { token } from "../../../../data/data";
+import { UserLogin } from "../../../../types/types";
 
 export const Login = () => {
-  const [email, setEmail] = useState<string>("");
-  const [password, setPassword] = useState<string>("");
+  const [userData, setUserData] = useState<UserLogin>({
+    email: "",
+    password: "",
+    isLoggedIn: false,
+  });
   const [error, setError] = useState<string>("");
 
+  // const { email, password } = useAppSelector((state) => state.user);
+  const { profiles } = useAppSelector((state) => state.user);
+  const dispatch = useAppDispatch();
+
   const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
     const { name, value } = e.target;
     if (name === "email") {
-      setEmail(value);
+      setUserData((prevUserData: UserLogin) => ({
+        ...prevUserData,
+        email: value,
+      }));
     } else if (name === "password") {
-      setPassword(value);
+      setUserData((prevUserData: UserLogin) => ({
+        ...prevUserData,
+        password: value,
+      }));
     }
   };
   const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault();
-    const user = profilesData.find((profile) => email === profile.email
-    && password === profile.password);
-    if (user) {
+    const userExists = profiles.find((profile) => userData.email === profile.email
+    && userData.password === profile.password);
+    if (userExists) {
+      const updatedUserData = {
+        ...userData,
+        isLoggedIn: true,
+      };
+      setUserData(updatedUserData);
```

Comment:
You don't need this line, since in [line 56](https://github.com/logo-cartt/logo-cart-ui/pull/4/files#diff-695b9e8d34ee8bd2e6a9c5b130cd91154dbcfd2f46d3399d451ce19258c580c3R56) you are resetting the user data to empty string.

The problem is that the setUserData is asynchronous react function.
It is better to just remove this setUserData(updatedUserData)
and then on the 56th line just write 
```
 setUserData({
   ...updatedUserData,
   email: "",
   password: "",
})
```

Reply:
done